src/systems/eigen_system.C

```diff
@@ -68,16 +70,17 @@ void EigenSystem::clear ()
   // Clear the parent data
   Parent::clear();
 
-  // Clean up the matrices
-  matrix_A.reset();
-  matrix_B.reset();
+  // The SparseMatrices are contained in _matrices
+  // and are cleared with the above call
+  matrix_A = nullptr;
+  matrix_B = nullptr;
+  precond_matrix = nullptr;
 
   // Operators
   shell_matrix_A.reset();
   shell_matrix_B.reset();
```

Comment:
Not to suggest touching them in this PR, but in the long run I can hardly wait see shell matrices turned into a third enum option for a first-class matrix.